### DISCUSSION TITLE
Dynamically control Faikin based on available power

### DISCUSSION THREAD

**Discussion by Tarudro** [revk/xESP32-Faikout#450] [Q&A]
@revk  Shoutout for this awesome project. You are a fucking legend for making this! I got my Daikin Prefera and everything worked like a charm.

However, I have a inquiry, to which I couldn't really find some good tactics yet.

I have some solar panels that I can monitor in Home Assistant, as well as my overall power consumption and grid connection.
I would like to achieve an automatic mode which I can enable that only starts the AC when there is spare electricity available and also scales the drawn current of the Daikin in a way, such that I don't need to get additional electricity from the grid, but can stay within my available electricity.

I'm not really sure what would be the best way to achieve this? Just write a script and dynamically change the target temperature? Is there a better way or some specific settings/commands that could help me to create a decent algorithm for this and scale the Daikin smoothly?

Cheers & thanks a lot

**Comment by revk** [comment]
This would be something outside the Faikin itself such as a script on HA, obviously. But the Faikin should allow you access to the controls you need to do something.

I think demand control may be the main one. Would be interested to hear what others had done with it.

If you have more than one unit together you may have to set the same demand control on all of them. There have been various discussions on this.

Good luck.

**Reply by Tarudro** [reply]
Thanks for the swift reply.
I only have one unit.

I saw the demand control and it’s discussions, especially with respect to multiple units. So far I couldn’t really make much of it but I‘ll give it a try and play around with it and see how much i can vary the power consumption.

I‘ll share the outcome here.

**Reply by 44616E** [reply]
@Tarudro 

I do this currently, with 6 units. But the same applies for 1. 
I use demand control when the available power is less than 1 AC (typically early morning). 

The ac control is pretty straight forward I just have a simple state machine for each unit and then the system as a whole. 

I do this with Node-Red with mosquito mqtt broker. Works well.

Adding solar forecasting will help too.

Just make sure you have logic / filters so that allow time for changes to take effect I.e. ac load can take a few minutes to build. But also solar can change rapidly so you don't want to adjust your ac because a single cloud went over head. 

Ping me if you want more info.
